backend/ScrubScreen.py

```diff
@@ -0,0 +1,135 @@
+
+    def __str__(self):
+        return self.name
+
+
+class MenuItem:
+    def __init__(self, name, Allergens):
+        self.name = name
+        self.Allergens = Allergens
+
+    def __str__(self):
+        return self.name
+
+
+class Allergen:
+    def __init__(self, name):
+        self.name = name
+
+    def __str__(self):
+        return self.name
+
+
+# 3 dining halls each with x stations. each station has x foodItems
+
+def ScrubScreen(Date):
+    # convert date into url
+    month = Date.strftime("%B")
+    day = Date.strftime("%d")
+    year = Date.strftime("%Y")
+    url = "https://hospitality.usc.edu/residential-dining-menus/?menu_date=" + \
+        month+"+"+day+"%2C"+year
+    # get the html from the website
+    html_text = req.get(url).text
+    # make it into soup
+    soup = BeautifulSoup(html_text, "lxml")
+    # get all meal blocks (brkfast, lunch, dinner etc)
+    MealTimes_HTML = soup.find_all("div", class_="hsp-accordian-container")
+    ListOfMealTime = []
+
+    for block in MealTimes_HTML:  # loop through all the meal blocks
+        # find the name of the block
+        MealTimeName = block.find(
+            "span", class_="fw-accordion-title-inner").text.split()[0]
+        # make a new list of all the dining halls in a block
+        DiningHalls_HTML = block.find_all("div", class_="col-sm-6 col-md-4")
+        ListOfDiningHalls = []
+
+        for Hall in DiningHalls_HTML:
+            DiningHallName = Hall.find("h3", class_="menu-venue-title").text
+            Station_Names_HTML = Hall.find_all("h4")
+            # the list of foods is a bunch of lists not under stations but under DiningHalls so have to get a list of foods in each stations here
+            MenuItems_ListList = Hall.find_all("ul")
+            ListOfStations = []
+
+            for i in range(len(Station_Names_HTML)):  # loop through all the stations
+                StationName = Station_Names_HTML[i].text  # station name
+                ListOfMenuItems = []
+                # when theres no items program bugs out; my workaround
+                if(StationName != "No items to display for this date"):
+                    MenuItems_HTML = MenuItems_ListList[i].find_all(
+                        "li")  # find all the food items
+                    for menuItem in MenuItems_HTML:  # iterate through all the food
+                        # get the name of the food item
+                        MenuItemName = menuItem.find(
+                            text=True, recursive=False)
+                        # get all the allergens for that food item
+                        Allergens_HTML = menuItem.find_all("i")
+                        ListOfAllergens = []
+                        for allergen in Allergens_HTML:  # add all the allergens for the food items
+                            AllergenName = allergen.find("span").text
+    # nest the lists inside eachother
+                            ListOfAllergens.append(Allergen(AllergenName))
```

Comment:
We can use a cool python feature called list comprehensions here to turn this into one line https://www.w3schools.com/python/python_lists_comprehension.asp

`ListOfAllergens = [allergen.find("span").text for allergen in Allergens_HTML]`

List comprehensions are fun and nice but they're not worth reducing code readability, so I'd only use it on a simple case like this